### ISSUE TITLE
[Bug]: Activation shortcut stops working upon switch to a new input source (and back)

### ISSUE THREAD

**Issue by jeutzsch** [quicksilver/Quicksilver#2943]
### Before submitting your bug report, please confirm you have completed the following steps

- [X] Searched the issue is not already listed in the [Issue List](https://github.com/quicksilver/Quicksilver/issues?q=is%3Aissue) (I've searched it)
- [X] Searched in the [Quicksilver Users' Group](https://groups.google.com/forum/#!forum/blacktree-quicksilver) for questions relating to tech support.
- [X] Downloaded the [latest version of Quicksilver](https://qsapp.com/) for your system and confirmed the issue still persists.

### Bug description

The key combination to activate QS (activation shortcut) stops working when I switch the input source using `command+space`, and still doesn't work when I switch back to the original input source. QS does _not_ crash in the process; it is still running, and I can access immediately it by clicking on the icon in the dock.

If I re-launch QS, the activation shortcut works again, until I switch the input source again. I first noticed this a few months ago, but I don't switch input sources that often, but it has happened often enough now that it seems to be a bug. I have been using QS for many years, and this is definitely a recent bug. Maybe it started with Ventura (OS 13), but I'm not certain.

Specifically, my activation shortcut is command+the-left-most-key-in-the-number-row (in other words, directly to the left of the number 1, and below esc). The input sources I use most are standard German (so the activation shortcut is `command+^`), Estonian (so the activation shortcut is `command+ˇ`) and a home-made layout based on standard German (so the activation shortcut is also `command+^`). As mentioned above, I have been switching between these input sources for many years while using QS, but this bug only appeared recently.

### Steps to reproduce

1. QS is running, and my activation shortcut activates QS
1. switch input source (keyboard layout) using `command+space` or by selecting a different source in the menu bar
1. QS activation shortcut no longer activates QS
1. switch back to the initial input source, and QS still isn't activated by the activation shortcut

### Expected behavior

I expect the activation shortcut to continue to activate QS, regardless of whether I switch input sources

### MacOS Version

Other

### Quicksilver Version

2.4.0 (4039)

### Crash Logs or Spindump

QS doesn't crash or hang, but here is a spindump (no idea if it's relevant):

[Spindump-QS.txt](https://github.com/quicksilver/Quicksilver/files/10460507/Spindump-QS.txt)

### Screenshots

(Doesn't seem relevant because there is nothing to see.)

### Additional info

Running Ventura 13.1 on a 14" MBP 2021 and on a MacMini 2018.

**Comment by jeutzsch** [comment]
Just a quick update to this bug ticket:
The issue still exists exactly as described above on the same devices with Quicksilver v2.4.1 (4040) running macOS 14.2 Sonoma.

**Comment by jeutzsch** [comment]
Still an issue. Even if it's connected to my system settings, there must be some way to avoid this happening across the board. But a couple weeks ago, I got annoyed enough at this, that I started using spotlight instead.
Note that I use the exact same key combination to trigger spotlight as I did for quicksilver and for switching keyboard layouts (see description above), but switching keyboard layouts has no effect at all on the key combination triggering spotlight.

**Comment by n8henrie** [comment]
Thanks for updating the issue!

So I can replicate, but I don't really think this is a Quicksilver bug.

> my activation shortcut is command+the-left-most-key-in-the-number-row

I presume you're referring to `` ` `` (the backtick)?

If so, keep in mind that your activation key is actually <kbd>cmd</kbd><kbd>`</kbd> (not the physical key to which that character is mapped, but the character itself).

When I add Spanish as a secondary keyboard layout, I can see quite clearly that the backtick is in the top left in my US layoutÑ

<img width="606" height="158" alt="Image" src="https://github.com/user-attachments/assets/1dc1e12d-d2d6-4de4-861f-b1879d4027a4" />

but it's over on the right (just to the left of `+`) when I'm using the Spanish layout:

<img width="590" height="161" alt="Image" src="https://github.com/user-attachments/assets/f69ea2c8-c54b-418e-8210-ab853ea553b4" />

So when I my English layout, cmd+"button to the left of `1`" works to invoke QS. When I change my input to Spanish that button stops working because it is no longer mapped to `` ` ``; instead, cmd + "button to the left of `+`" works.

Does that make sense?

The only workaround that I can think of is to use a key that stays the same between both keyboard layouts (maybe <kbd>cmd</kbd><kbd>esc</kbd>?). Perhaps you can something like [hidutil](https://rakhesh.com/mac/using-hidutil-to-map-macos-keyboard-keys/) to change the key's behavior? Or repurpose something like caps lock (I remap caps to something like F19 to function as a "hyper" key).

Hope this is helpful. Will plan to close this issue shortly unless it looks like this is really a QS bug. And thanks again for the detailed bug report!

**Comment by jeutzsch** [comment]
Thanks for the response, @n8henrie!

What you describe makes sense (that invoking QS is triggered by a specific character mapping, and not to a physical key, so when I change to a different layout, I need to find that character on the new layout in order to invoke QS. By the way, yes, I am referring to the physical key which has ` ̀` (backtick or grave accent?) on the standard US layout, but is a `^` (circumflex accent) on the standard German layout.

The reason I think that it still might be a QS bug is that when I switch to another layout, and then switch *back* to my main layout (German), I can't invoke QS any more using my key combination, even though that key still inputs `^` (for example in a text editor, or in the [UnicodeChecker app](https://earthlingsoft.net/UnicodeChecker/index.html)).

But since I seem to be the only one with the problem, maybe it's not a QS bug but instead some strange settings on my specific set up ... unfortunately I have no idea how to trouble-shoot that. And in the meantime, I've (regrettably) transitioned to spotlight anyway, so if you decide to close this issue I will certainly understand.

**Comment by n8henrie** [comment]
> The reason I think that it still might be a QS bug is that when I switch to another layout, and then switch back to my main layout (German), I can't invoke QS any more using my key combination

Interesting. I'd like to dig in a little more!

Just re-confirming my prior findings...

- [x] change QS from my default of <kbd>ctrl</kbd><kbd>space</kbd> to <kbd>⌘</kbd><kbd>`</kbd>
- [x] settings -> Keyboard -> Input Sources (`edit`)
- [x] check show input source in menu bar
- [x] `+` in bottom left of same window
- [x] add Spanish
- [x] confirm that <kbd>⌘</kbd><kbd>`</kbd> brings up QS while keyboard is in U.S. mode
- [x] confirm that <kbd>ctrl</kbd><kbd>space</kbd> changes my keyboard into Spanish mode
- [x] confirm that <kbd>⌘</kbd><kbd>\`</kbd> (using the physical key with the `` ` ``) **does not** bring up QS while keyboard is in Spanish mode
- [x] confirm that <kbd>⌘</kbd><kbd>[</kbd> (using the physical key with the `` [ ``), just to the right of `p` **does indeed** bring up QS while keyboard is in Spanish mode
- [x] confirm that <kbd>ctrl</kbd><kbd>space</kbd> changes my keyboard back into U.S. mode
- [x] confirm that <kbd>⌘</kbd><kbd>`</kbd> still brings up QS with keyboard back in U.S. mode

EDIT: Sorry for the premature submit. Too many modifier keys going on 😆

So in short I can't reproduce your issue. Could you submit a few screenshots showing the name of the alternative keyboard and its key layout (like I did for my Spanish example above)?
